jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1949,4 +1962,43 @@ private static void failed(Callback callback, Throwable failure)
             throw t;
         }
     }
+
+    private static class ThreadPoolWrapper implements ThreadPool
+    {
+        private final Executor _executor;
+
+        private ThreadPoolWrapper(Executor executor)
+        {
+            _executor = executor;
+        }
+
+        @Override
+        public void execute(Runnable command)
+        {
+            _executor.execute(command);
+        }
+
+        @Override
+        public void join()
+        {
+        }
```

Comment:
```suggestion
        public void join()
        {
            if (_executor instanceof ThreadPool threadpool)
                threadpool.join();
        }
```

Reply:
@gregw I know already this is not a `ThreadPool`, see where this wrapper is created.